iterating-over-data/iterating-and-special-cases.md

```diff
@@ -0,0 +1,132 @@
+
+Are we able to stop a loop before it gets to the last element? Are we able to use `for` loops on other data structures?
+
+By learning these different pieces of syntax, we can learn to wield them wisely whenever it becomes relevant.
+
+## Vocabulary and Synonyms
+
+| Vocab      | Definition                                                       | Synonyms | How to Use in a Sentence                                                                     |
+| ---------- | ---------------------------------------------------------------- | -------- | -------------------------------------------------------------------------------------------- |
+| `break`    | A keyword that will exit an entire loop                          | -        | "We can break out of the loop using `break`," "The loop exited early because it saw `break`" |
+| `continue` | A keyword that will immediately advance one interation in a loop | -        | "The loop moved on because it saw `continue`"                                                |
+| `range()`  | A Python function that creates a sequence                        | -        | "We used a `for` loop and `range()` in order to iterate through a series of numbers.         |
+
+## `break` Exits a Loop
+
+`break` is a keyword that we can put in a `for` loop. When our code comes across `break`, it will exit the `for` loop entirely.
+
+```python
+for fruit in ["apples", "oranges", "bananas"]:
+    print("I'm in the for loop! The value of fruit is:", fruit)
+    break
+    print("This message will never print, because the for loop saw break before it saw me...")
+```
+
+We can pair this with conditionals (and other logic)!
+
+```python
+print("I'm looking for a four leaf clover in the field...")
+
+for field_item in ["grass", "grass", "more grass", "four-leaf clover", "rocks", "rocks", "more rocks"]:
+    print("...")
+    if field_item == "four-leaf clover":
+        print("I found a four-leaf clover!")
+        print("My hunt is over! I can leave now")
+        break
+    print(f"I found {field_item} in the field...")
+    print("Better keep looking.")
+```
+
+### !callout-info
+
+## for-else
+
+Turns out we can pair a `for` loop with an `else` clause! The `else` clause will run after a `for` loop finishes **if** it did **not** encounter a `break`. The `else` is on the same level of indentation as the `for`.
+
+### !end-callout
+
+```python
+for field_item in ["grass", "grass", "more grass", "rocks", "rocks", "more rocks"]:
+    print("...")
+    if field_item == "four-leaf clover":
+        print("I found a four-leaf clover!")
+        print("My hunt is over! I can leave now")
+        break
+    print(f"I found {field_item} in the field...")
+    print("Better keep looking.")
+else:
+    print("I didn't find a four-leaf clover today.")
+```
+
+## `continue` Advances One Loop
+
+`continue` is another keyword we can put in a `for` loop. When our code sees `continue`, it will immediately move to the next iteration.
```

Comment:
avoid perspective shift in description of break.
shift voicing to the interpreter hitting the break, rather than our code.
provide a bit more explicit description of what the continue does.

```suggestion
`continue` is another keyword that can be used in a `for` loop. When the Python interpreter encounters a `continue`, it will immediately jump to the beginning of the current loop, skipping any instructions from after the `continue` until the end of the loop. The loop will resume execution with the next item in the iteration, if one exists. Otherwise, the loop will terminate.
```